cmd/internal/trender.go

```diff
@@ -28,9 +30,27 @@ This will render a provided manifest instance-group
 		instanceGroupName := viper.GetString("instance_group_name")
 		address := viper.GetString("address")
 		az := viper.GetString("az")
+		if len(az) == 0 {
+			return fmt.Errorf("az cannot be empty")
+		}
 		id := viper.GetString("id")
-		index := viper.GetInt("index")
+		index := viper.GetString("index")
+		if len(index) == 0 {
+			// calculate index
+			azIndex, err := strconv.Atoi(os.Getenv("CF_OPERATOR_AZ_INDEX"))
+			if err != nil {
+				return err
+			}
+			podIndex, err := strconv.Atoi(os.Getenv("POD_INDEX"))
+			if err != nil {
+				return fmt.Errorf("podIndex not found")
+			}
+			index = strconv.Itoa(podIndex * azIndex)
```

Comment:
Where does this index calculation algorithm come from? There will be conflicts, e.g. [azIndex: 0, podIndex: 1] and [azIndex: 1, podIndex: 0] would both become index 0. Is that a problem?

Reply:
@aduffeck Yes. I didn't think of this. This is stated in [Link](https://github.com/cloudfoundry-incubator/cf-operator/blob/58671dbe9a7e910d859f46c17042f89fed054032/docs/rendering_templates.md).  @viovanov can answer maybe.